mat64/matrix.go

```diff
@@ -22,6 +22,59 @@ type Matrix interface {
 	// At returns the value of a matrix element at (r, c). It will panic if r or c are
 	// out of bounds for the matrix.
 	At(r, c int) float64
+
+	// T returns the transpose of the Matrix. The manner in which this is implemented
+	// is implementation dependent, including if T returns a copy of the underlying
```

Comment:
I don't see the need for "The manner in which this is implemented is implementation dependent"; it is tautological. The only thing that needs to be conveyed is whether the transpose will reflect changes to the original matrix after the call to T. I think the original wording I have above does this. I also thing "if" reads less well than "whether".